In [3]:
# import relevant libraries

import pandas as pd

import warnings
warnings.filterwarnings('ignore')

In [4]:
# read the two datasets of 2019 - Open Athletes & Open Scores
df_19_ath = pd.read_csv('./data/2019_opens_athletes.csv')
df_19_sco = pd.read_csv('./data/2019_opens_scores.csv')

In [6]:
df_19_ath.head()

competitorid     competitorname firstname    lastname postcompstatus  \
0          2536    Samantha Briggs  Samantha      Briggs       accepted   
1        485089    Renata Pimentel    Renata    Pimentel       accepted   
2         16973    Carleen Mathews   Carleen     Mathews            NaN   
3        751083  Danila Capaccetti    Danila  Capaccetti            NaN   
4        313257        Hope Cicero      Hope      Cicero            NaN   

  gender          profilepics3key countryoforigincode countryoforiginname  \
0      F   0e63d-P2536_14-184.jpg                  GB      United Kingdom   
1      F  04e97-P485089_5-184.jpg                  BR              Brazil   
2      F   b663a-P16973_6-184.jpg                  US       United States   
3      F                pukie.png                  IT               Italy   
4      F  f204b-P313257_1-184.jpg                  US       United States   

   divisionid  affiliateid          affiliatename  age  height  weight  \
0          19         4098    CrossFit Black Five   37    1.70   61.23   
1          19        15868        CrossFit Gurkha   36    1.74   73.00   
2          19        10471  CrossFit Saint Helens   35    1.57   62.14   
3          19         9329   CrossFit Black Shark   35    1.70   71.00   
4          19          438      CrossFit Billings   36    1.55   61.23   

   overallrank  overallscore  is_scaled       division  
0            1            33          0  Women (35-39)  
1            2            66          0  Women (35-39)  
2            3           101          0  Women (35-39)  
3            4           139          0  Women (35-39)  
4            5           176          0  Women (35-39)

* some features may be neglected (?):
  - competitorname
  - firstname
  - lastname
  - postcompstatus
  - profilepics3key
  - countryoforigincode (name already exists)
  - affiliatename (affiliateid has the same information)
* has divisionid same information as division?
* we need competitorid to combine both dataframes later

In [13]:
df_19_sco.head()

affiliate                                          breakdown  \
0  CrossFit RDU                   9 rounds +\n10 wall-ball shots\n   
1  CrossFit RDU  Within 16 minutes:\n3 rounds +\n25 toes-to-bar...   
2  CrossFit RDU  200-ft. OH lunge\n50 box step-ups\n50 strict H...   
3  CrossFit RDU                                 132 reps\n6 rounds   
4  CrossFit RDU                                           210 reps   

   competitorid     division  is_scaled           judge  ordinal  rank  \
0         96511  Men (45-49)          0     Erin Miller        1     1   
1         96511  Men (45-49)          0  Harper Thorsen        2     4   
2         96511  Men (45-49)          0  Harper Thorsen        3     1   
3         96511  Men (45-49)          0  Harper Thorsen        4    36   
4         96511  Men (45-49)          0  Harper Thorsen        5     1   

   scaled     score scoredisplay       scoreidentifier   time  
0       0  13520000     352 reps  27f30f9a8c0a564ae799    NaN  
1       0  13420368     342 reps  0ed3d1264f25a8f1890d    NaN  
2       0  11800018         9:42  f2a143399a330c95321b  582.0  
3       0  11320009        11:51  89101e401c6c85997363  711.0  
4       0  12100573        10:27  f7588c9174f1fe90f5c4  627.0

* what is scoreidentifier?

In [8]:
df_19_ath.shape

(572653, 19)

In [9]:
df_19_sco.shape

(2863265, 13)

* 572,653 athletes with 5 competition workouts per athlete results in 2,863,265 scores observations

In [5]:
df_19_ath.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 572653 entries, 0 to 572652
Data columns (total 19 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   competitorid         572653 non-null  int64  
 1   competitorname       572653 non-null  object 
 2   firstname            572653 non-null  object 
 3   lastname             572653 non-null  object 
 4   postcompstatus       302 non-null     object 
 5   gender               572653 non-null  object 
 6   profilepics3key      572653 non-null  object 
 7   countryoforigincode  572348 non-null  object 
 8   countryoforiginname  572653 non-null  object 
 9   divisionid           572653 non-null  int64  
 10  affiliateid          572653 non-null  int64  
 11  affiliatename        540821 non-null  object 
 12  age                  572653 non-null  int64  
 13  height               304823 non-null  float64
 14  weight               323754 non-null  float64
 15  overallrank      

* height and weight has many missing values

In [10]:
df_19_sco.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2863265 entries, 0 to 2863264
Data columns (total 13 columns):
 #   Column           Dtype  
---  ------           -----  
 0   affiliate        object 
 1   breakdown        object 
 2   competitorid     int64  
 3   division         object 
 4   is_scaled        int64  
 5   judge            object 
 6   ordinal          int64  
 7   rank             int64  
 8   scaled           int64  
 9   score            int64  
 10  scoredisplay     object 
 11  scoreidentifier  object 
 12  time             float64
dtypes: float64(1), int64(6), object(6)
memory usage: 284.0+ MB
